### ISSUE TITLE
New "Argument type is partially unknown" error in 1.1.247

### ISSUE THREAD

**Issue by kurtschelfthout** [microsoft/pyright#3505]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
An "Argument type is partially unknown" error started appearing in 1.1.247. As far as I understand the code should typecheck.

**To Reproduce**

Typecheck the code below

**Expected behavior**
No type errors. 

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
T = TypeVar("T")

class Shrink(Protocol[T]):
    def __call__(self, value: T) -> Iterable[T]:
        ...

@dataclass(frozen=True)
class CandidateTree(Generic[T]):
    value: T
    candidates: Iterable[CandidateTree[T]]

def tree_from_shrink(value: T, shrink: Shrink[T]) -> CandidateTree[T]:
    return CandidateTree(
        value = value,
        candidates = (
            tree_from_shrink(v, shrink)
            for v in shrink(value))
        )
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
VsCode w/ PyLance 2022.5.2

**Additional context**

**Comment by erictraut** [comment]
Thanks for reporting. This is a duplicate of https://github.com/microsoft/pyright/issues/3481, so I'm going to close this one.
